src/onegov/core/upgrade.py

```diff
@@ -482,17 +482,35 @@ def has_column(self, table: str, column: str) -> bool:
             table, schema=self.schema
         )}
 
-    def has_enum(self, enum: str) -> bool:
-        return self.session.execute(f"""
+    def has_constraint(self, table_name: str, constraint_name: str) -> bool:
+        return self.session.execute(text("""
+            SELECT EXISTS (
+                SELECT 1 FROM information_schema.table_constraints
+                WHERE table_schema = :schema
+                  AND table_name = :table_name
+                  AND constraint_type = 'FOREIGN KEY'
```

Comment:
```suggestion
                  AND constraint_type = 'FOREIGN KEY'
```
This condition I'd either take out or make it an argument to the function, since we may want to check other kinds of constraints, like `UNIQUE` constraints.